deepinterpolation/cli/inference.py

```diff
@@ -0,0 +1,77 @@
+import argschema
+import json
+import h5py
+import numpy as np
+from pathlib import Path
+
+from deepinterpolation.cli.schemas import InferenceInputSchema
+from deepinterpolation.generic import ClassLoader
+
+
+class Inference(argschema.ArgSchemaParser):
+    default_schema = InferenceInputSchema
+
+    def run(self):
+        self.logger.name = type(self).__name__
+
+        outdir = Path(self.args['inference_params']['output_file']).parent
+        if self.args["output_full_args"]:
+            full_args_path = outdir / "inference_full_args.json"
+            with open(full_args_path, "w") as f:
+                json.dump(self.args, f, indent=2)
+            self.logger.info(f"wrote {full_args_path}")
+        uid = self.args['run_uid']
+
+        # save the json parameters to 2 different files
+        inference_json_path = outdir / f"{uid}_inference.json"
+        with open(inference_json_path, "w") as f:
+            json.dump(self.args['inference_params'], f,  indent=2)
+        self.logger.info(f"wrote {inference_json_path}")
+
+        generator_json_path = outdir / f"{uid}_generator.json"
+        with open(generator_json_path, "w") as f:
+            json.dump(self.args['generator_params'], f, indent=2)
+        self.logger.info(f"wrote {generator_json_path}")
+
+        generator_obj = ClassLoader(generator_json_path)
+        data_generator = generator_obj.find_and_build()(generator_json_path)
+
+        inferrence_obj = ClassLoader(inference_json_path)
+        inferrence_class = inferrence_obj.find_and_build()(
+            inference_json_path,
+            data_generator)
+
+        self.logger.info("created objects for inference")
+        inferrence_class.run()
+
+        # patch up the output movie
+        # This code below will go within the inference library as pre/post
+        # processings modules. Adding temporary fix to remove for non-h5 files
+        # so that the CLI works with tiff, dat, ... files.
+        if '.h5' in self.args["generator_params"]["train_path"]:
```

Comment:
@jeromelecoq it would make more sense to me to keep the scope of this PR narrow as "CLI for ophys h5 inputs", fix the normalization and units of the inference module, remove this, then, support these other file formats, rather than adding temporary features that we don't actually need right now.

Reply:
Some labs doing ophys are working with tiff files already so I was reviewing this in the context of a broader use of the CLI then just our internal use. I think we will have to make these fixes. I am fine with doing them in a following PR if we are doing them.

Reply:
let's just leave it here for now, since you already did it, but, let's try to keep these PRs very narrow scoped going forward.
the idea is to build up a bunch of small changes and then, when we are somewhat content, release a version, and then advertise that version as having some functionality to be generally useful